packaging/os/apk.py

```diff
@@ -0,0 +1,219 @@
+
+def query_package(module, name):
+    cmd = "apk -v info --installed %s" % (name)
+    rc, stdout, stderr = module.run_command(cmd, check_rc=False)
+    if rc == 0:
+        return True
+    else:
+        return False
+
+def query_latest(module, name):
+    cmd = "apk version %s" % (name)
+    rc, stdout, stderr = module.run_command(cmd, check_rc=False)
+    search_pattern = "(%s)-[\d\.\w]+-[\d\w]+\s+(.)\s+[\d\.\w]+-[\d\w]+\s+" % (name)
+    match = re.search(search_pattern, stdout)
+    if match and match.group(2) == "<":
+        return False
+    return True
+
+def upgrade_packages(module):
+    if module.check_mode:
+        cmd = "apk upgrade --simulate"
+    else:
+        cmd = "apk upgrade"
+    rc, stdout, stderr = module.run_command(cmd, check_rc=False)
+    if rc != 0:
+        module.fail_json(msg="failed to upgrade packages")
+    if re.search('^OK', stdout):
+        module.exit_json(changed=False, msg="packages already upgraded")
+    module.exit_json(changed=True, msg="upgraded packages")
+
+def install_packages(module, names, state):
+    upgrade = False
+    uninstalled = []
+    for name in names:
+        if not query_package(module, name):
+            uninstalled.append(name)
+        elif state == 'latest' and not query_latest(module, name):
+            upgrade = True
+    if not uninstalled and not upgrade:
+        module.exit_json(changed=False, msg="package(s) already installed")
+    names = " ".join(uninstalled)
+    if upgrade:
+        if module.check_mode:
+            cmd = "apk add --upgrade --simulate %s" % (names)
+        else:
+            cmd = "apk add --upgrade %s" % (names)
+    else:
+        if module.check_mode:
+            cmd = "apk add --simulate %s" % (names)
+        else:
+            cmd = "apk add %s" % (names)
+    rc, stdout, stderr = module.run_command(cmd, check_rc=False)
+    if rc != 0:
+        module.fail_json(msg="failed to install %s" % (names))
+    module.exit_json(changed=True, msg="installed %s package(s)" % (names))
+
+def remove_packages(module, names):
+    installed = []
+    for name in names:
+        if query_package(module, name):
+            installed.append(name)
+    if not installed:
+        module.exit_json(changed=False, msg="package(s) already removed")
+    names = " ".join(installed)
+    if module.check_mode:
+        cmd = "apk del --purge --simulate %s" % (names)
+    else:
+        cmd = "apk del --purge %s" % (names)
+    rc, stdout, stderr = module.run_command(cmd, check_rc=False)
+    if rc != 0:
+        module.fail_json(msg="failed to remove %s package(s)" % (names))
+    module.exit_json(changed=True, msg="removed %s package(s)" % (names))
+        
+# ==========================================
+# Main control flow.
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            state = dict(default='present', choices=['present', 'installed', 'absent', 'removed', 'latest']),
+            name = dict(type='str'),
```

Comment:
use `type='list'` here, so users can pass a list like this:
~~~
- apk: name={{ item }}
  with_items:
  - foo
  - bar
~~~
or even comma separated as well (out of the box ansible magic)!
~~~
- apk: name=foo,bar
~~~

In your code, you will get python list to use, so your code must be adjusted accordantly.